Automattic/wp-calypso
# Checkout: block multisite incompatible product purchases

#### Changes proposed in this Pull Request

* We want to block product purchases that are incompatible with multisite. Currently, this means blocking Jetpack Backup and Jetpack Scan purchases.

#### Testing instructions

Prerequisite: self-hosted Jetpack multisite.

* Run this PR.
* Select your Jetpack multisite.
* Visit `http://calypso.localhost:3000/checkout/[site]/jetpack_backup`.
* Verify that you see a Notice/Banner at the top of the card component that says you can't buy Jetpack Backup.
* Verify that the purchase button is disabled.
* Visit `http://calypso.localhost:3000/checkout/[site]/jetpack_daily`.
* Verify that you see a Notice/Banner at the top of the card component that says you can't buy Jetpack Backup Daily.
* Verify that the purchase button is disabled.
* Visit `http://calypso.localhost:3000/checkout/[site]/jetpack_scan`.
* Verify that you see a Notice/Banner at the top of the card component that says you can't buy Jetpack Scan.
* Verify that the purchase button is disabled.

(Try these steps with as much types of payments as you can).

**Note: to test AliPay and WeChat payment types, you need to sandbox D45446-code and D45650-code.**

Fixes 1184358343400545-as-1184719508034177

#### Demo
##### Payment type: credits
![image](https://user-images.githubusercontent.com/3418513/87679476-9d635580-c752-11ea-9fa7-6ba45ac31b6c.png)

##### Payment type: credit card with Stripe
![image](https://user-images.githubusercontent.com/3418513/87686103-81fc4880-c75a-11ea-9a59-b19b16343639.png)

##### Payment type: PayPal
![image](https://user-images.githubusercontent.com/3418513/87685056-5462cf80-c759-11ea-8b94-0e0c7b8b66ae.png)

##### Payment type: AliPay
![image](https://user-images.githubusercontent.com/3418513/87710842-c26cbe00-c77c-11ea-8d68-d4f662e6e94b.png)

##### Payment type: WeChat
![image](https://user-images.githubusercontent.com/3418513/87711778-204dd580-c77e-11ea-93f4-7549903f6858.png)
